subsys/power/device.c

```diff
@@ -160,7 +160,8 @@ void sys_pm_create_device_list(void)
 		const struct device *dev = &all_devices[pmi];
 
 		/* Ignore "device"s that don't support PM */
-		if (dev->device_pm_control == device_pm_control_nop) {
+		if ((dev->device_pm_control == NULL) ||
+		    (dev->device_pm_control == device_pm_control_nop)) {
```

Comment:
If we would replace all `device_pm_control_nop` with `NULL`, then there is no need to check for two cases here.